exchanges/coinbasepro/coinbasepro.go

```diff
@@ -224,9 +225,8 @@ func (c *CoinbasePro) GetHistoricRates(ctx context.Context, currencyPair, start,
 		values.Set("end", "")
 	}
 
-	allowedGranularities := [6]int64{60, 300, 900, 3600, 21600, 86400}
-	validGran, _ := common.InArray(granularity, allowedGranularities)
-	if !validGran {
+	allowedGranularities := []int64{60, 300, 900, 3600, 21600, 86400}
```

Comment:
NOTE:  Just had a quick play and which I thought was interesting, we can reduce the two parameter instance of a lot of these. Out of scope though. 

```go
// Container is a generic type for slices
type Container[T comparable] []T

func (c Container[T]) Contains(in T) bool {
	return slices.Contains(c, in)
}

func TestXxx(t *testing.T) {
	allowedGranularities := Container[int64]{60, 300, 900, 3600, 21600, 86400}
	assert.True(t, allowedGranularities.Contains(300))
}
```